Source/Diagnostics/ReducedDiags/FieldReduced.cpp

```diff
@@ -0,0 +1,258 @@
+/* Copyright 2019-2020 Neil Zaim, Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "FieldReduced.H"
+#include "WarpX.H"
+
+using namespace amrex;
+
+// constructor
+FieldReduced::FieldReduced (const std::string& rd_name, const std::vector<std::string>& field_type)
+: ReducedDiags{rd_name}
+{
+
+    // RZ coordinate is not working
+#if (defined WARPX_DIM_RZ)
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+        "FieldReduced reduced diagnostics does not work for RZ coordinate.");
+#endif
+
+    for (int i = 0; i < field_type.size(); ++i)
+    {
+        std::string type = field_type[i];
+
+        if (type.compare("FieldEnergy") == 0) {m_fieldEnergy = true;}
+        else if (type.compare("MaxField") == 0) {m_maxField = true;}
+        else { Abort("No matching type found for reduced field diagnostic."); }
+    }
+
+    // read number of levels
+    int nLevel = 0;
+    ParmParse pp("amr");
+    pp.query("max_level", nLevel);
+    nLevel += 1;
+
+    // resize data array
+    int data_size = 0;
+    if (m_fieldEnergy) {data_size += 3*nLevel;}
+    if (m_maxField)
+    {
+        m_offset_maxField = data_size;
+        data_size += 8*nLevel;
+    }
+    m_data.resize(data_size,0.0);
+
+    if (ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            // open file
+            std::ofstream ofs;
+            ofs.open(m_path + m_rd_name + "." + m_extension,
+                std::ofstream::out | std::ofstream::app);
+            // write header row
+            ofs << "#";
+            ofs << "[1]step()";
+            ofs << m_sep;
+            ofs << "[2]time(s)";
+
+            if (m_fieldEnergy)
+            {
+                for (int lev = 0; lev < nLevel; ++lev)
+                {
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(3+3*lev) + "]";
```

Comment:
the numbers here would ideally also get names and/or could reuse at least for the `3*` the constant above :)